onnxruntime/core/providers/openvino/openvino_graph.cc

```diff
@@ -0,0 +1,464 @@
+                                           size_t batch_slice_idx,
+                                           size_t infer_req_idx) {
+
+  auto infer_request = infer_requests_[infer_req_idx];
+
+  // Wait for Async inference completion
+  infer_request->Wait(InferenceEngine::IInferRequest::WaitMode::RESULT_READY);
+  auto graph_output_info = openvino_network_->getOutputsInfo();
+
+  size_t i = 0;
+  for (auto output_info_iter = graph_output_info.begin();
+       output_info_iter != graph_output_info.end(); ++output_info_iter, ++i) {
+    // Get OpenVINO's output blob
+    auto graph_output_blob = infer_request->GetBlob(output_info_iter->first);
+    auto graph_output_buffer =
+        graph_output_blob->buffer().as<InferenceEngine::PrecisionTrait<InferenceEngine::Precision::FP32>::value_type*>();
+
+    size_t output_data_size = graph_output_blob->byteSize();
+    char* tensor_data = ort.GetTensorMutableData<char>(output_tensors[i]);
+    char* batch_memory_offset = tensor_data + output_data_size * batch_slice_idx;
+
+    // Copy output results back to ONNX-RT's output buffers
+    std::memcpy(batch_memory_offset, graph_output_buffer, output_data_size);
+  }
+}
+
+void OpenVINOGraph::GetInputTensors(Ort::CustomOpApi ort, OrtKernelContext* context, const OrtValue* input_tensors[]) {
+  size_t input_count = openvino_network_->getInputsInfo().size();
+
+  for (size_t i = 0; i < input_count; i++) {
+    input_tensors[i] = ort.KernelContext_GetInput(context, input_indexes_[i]);
+  }
+}
+
+void OpenVINOGraph::GetOutputTensors(Ort::CustomOpApi ort, OrtKernelContext* context, OrtValue* output_tensors[], size_t batch_size) {
+  auto graph_output_info = openvino_network_->getOutputsInfo();
+
+  // All infer_requests process identical tensor slices from the batch.
+  // So using info from first infer_request to allocate all output tensors.
+  auto infer_request = infer_requests_[0];
+
+  size_t i = 0;
+  for (auto output_info_iter = graph_output_info.begin();
+       output_info_iter != graph_output_info.end(); ++output_info_iter, ++i) {
+    auto graph_output_blob = infer_request->GetBlob(output_info_iter->first);
+    auto graph_output_dims = graph_output_blob->getTensorDesc().getDims();
+
+    if (batch_size > 1) {
+      // Add the batch size as dim 0.
+      graph_output_dims.insert(graph_output_dims.begin(), batch_size);
+    }
+
+    size_t num_dims = graph_output_dims.size();
+    int64_t output_shape[num_dims];
+    for (size_t j = 0; j < num_dims; j++) {
+      output_shape[j] = (int64_t)graph_output_dims[j];
+    }
+
+    output_tensors[i] = ort.KernelContext_GetOutput(context, i, output_shape, num_dims);
+  }
+}
+
+void OpenVINOGraph::Infer(Ort::CustomOpApi ort, OrtKernelContext* context) {
+  // Preliminary Thread safety mechanism
+  // Currently allows only one Infer execution at a time
+  std::lock_guard<std::mutex> lock(compute_lock_);
+
+  std::cout << "[OpenVINO-EP] Inference Started\n";
```

Comment:
i hope we're going to remove all the cout/cerr.

Reply:
Agreed. Logging mechanism with verbose option is being implemented, that will remove these statements.